elastic/kibana
# [Synthetics] ensure monitor namespace is kibana space

## Summary
Resolves https://github.com/elastic/kibana/issues/128820

1. Automatically uses a fleet-compliant representation of the Kibana space id as the namespace for the monitor configuration
2. Formats the raw Kibana space id to a fleet-compliant format
3. Automatically populates the namespace field in the Synthetics UI with the appropriate namespace
![image](https://user-images.githubusercontent.com/11356435/190205437-bf062a5e-b0e7-4faf-9bb6-abd9a59a1bd4.png)
4. Includes validation for the namespace in the Synthetics UI form
![image](https://user-images.githubusercontent.com/11356435/190205462-628ab651-4a42-41fc-b645-d54a275e94e6.png)
5. Preserves legacy Uptime behavior by including `preserve_namespace` query param on create monitor api.


### Checklist



### Testing

1. Create a new space, ideally with a Kibana space id that has `-` in it. Basically, you can do this by creating a space with at least two words in the name, like `Test Space`.
2. Navigate to Synthetics UI create monitor flow and inspect the namespace field. The field should contain the Kibana space id, with `-` replaced by `_`
3. Save the monitor. Ensure that the resulting documents appear in that space.
